sync/syncer.go

```diff
@@ -287,9 +288,29 @@ func (s *Syncer) Synchronise() {
 	s.handleNotSynced(currentSyncLayer)
 }
 
+func (s *Syncer) FastValidation(block *types.Block) error {
```

Comment:
Why is it a method of syncer? why not inject it in the constuctor?

Reply:
i think its more readable this way syncer fits into the ValidationInfra interface that holds all methods for validation queue 
seams cleaner to me, it should also be private ill change it

Reply:
Please rethink it. Having a method memeber gives you two main adbvantages:
1. Easier to test (just replace the member func with whatever mock you like)
2. It unburdens the syncer from this extra knowledge. It is pretty simple to say "I want to validate a block so I have a validation function" than "I want to validate a block so I keep all the components required so I can use them when I need to validate".

Regarding the validation infra - there is no wrong with having an internal struct for it (if the interface really requires just validation) or simply passing the validationFunc as before. IMO, You should try to avoid adding extra knowledge/abilities to the syncer and break it to (interface) components that tend to be smaller and simpler.

Reply:
its not that i disagree with your points 
its just that im aiming for this to be changed soon and building on top of it will make it harder in the next pr :)